Kernel/include/kernelTypes.h

```diff
@@ -47,7 +47,7 @@ typedef struct {
 /**
  * @brief Represents a semaphore.
  */
-typedef int TSem;
+typedef uint8_t Tsem;
```

Comment:
Might be a better idea to use an `int`, as to be able to indicate errors (like with `TPid`).

This will solve issues in `sem_open()` where on errors you try to return a negative value, but the negative integer value will be interpreted as an `uint8_t`.